tools/pnpm-deps-gen.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# First regex can be read as the following:
+# .*\.pnpm\/(@?[^@]+).*
+# We use the literal .pnpm to find the start of the package name
+# After that, we use the pattern "@?[^@]+" to search for the name
+# that appears first, instead of ".+".
+#
+# Sometimes we could have foo@1.0.0_bar@1.0.0, in which case we want
+# to capture foo. With ".+", we could capture foo@1.0.0_bar.
+# Instead, we search for the string of characters that do not include
+# "@", which translates to "[^@]+". However, some package names include
+# @ at the start, so we cover that case, as well.
+
+option=$1
+filename=$2
+
+if [[ $# -eq 2 && $option = "-o" && $filename ]]; then
+  echo "Start collecting dependencies"
+  pnpm list -r --parseable --depth=Infinity | sed -n "s/.*\.pnpm\/\(@\?[^@]\+\).*/\1/p" | sort -u | sed "s/+/\//" > $filename
```

Comment:
If you want to make this more readable, you can use extended regular expression and choose a custom delimiter to avoid escaping characters.
```suggestion
  pnpm list -r --parseable --depth=Infinity | sed -n -r -e "s-.*\.pnpm/(@?[^@]+).*-\1-p" | sort -u | sed "s/+/\//" >  $filename
```

  Also, are we pursuing depth=Infinity, seems too deep to me. The whole script took 20s to execute.
cc @humphd